### PR TITLE
Added metavar and choices to --state flag in airflow dags list-jobs CLI for suggesting valid state arguments.

### PR DESCRIPTION
**Pull request by rohan472000**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/apache-airflow/administration-and-deployment/security/audit-logs.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
+Audit Logs in Airflow
```
Comment (potiuk) [inline docs/apache-airflow/administration-and-deployment/security/audit-logs.rst]: This is accidental I guess?

Comment (rohan472000) [reply docs/apache-airflow/administration-and-deployment/security/audit-logs.rst]: yeah..it was merged in #30405

File: airflow/cli/cli_config.py [hunk 2]
```diff
@@ -277,7 +277,12 @@ def string_lower_type(val):
 ARG_NO_BACKFILL = Arg(
     ("--no-backfill",), help="filter all the backfill dagruns given the dag id", action="store_true"
 )
-ARG_STATE = Arg(("--state",), help="Only list the dag runs corresponding to the state")
+ARG_STATE = Arg(
+    ("--state",),
+    help="Only list the dag runs corresponding to the state",
+    metavar="(table, json, yaml, plain)",
+    choices=("table", "json", "yaml", "plain"),
+)
```
Comment (uranusjr) [inline airflow/cli/cli_config.py]: The choices seem wrong…?

Comment (rohan472000) [reply airflow/cli/cli_config.py]: didn't got u,please elaborate.

Comment (hussein-awala) [reply airflow/cli/cli_config.py]: you just copied the choices from `ARG_OUTPUT`.
The choices for the dag run state are `running`, `queued`,  `success` and `failed`

Comment (hussein-awala) [reply airflow/cli/cli_config.py]: IMO it's better to add them dynamically:
```python
>>> from airflow.utils.state import DagRunState
>>> tuple(state.value for state in DagRunState)
('queued', 'running', 'success', 'failed')
>>> str(tuple(state.value for state in DagRunState))
"('queued', 'running', 'success', 'failed')"
```
@uranusjr wdyt?

Comment (rohan472000) [reply airflow/cli/cli_config.py]: plz check now

File: airflow/cli/cli_config.py [hunk 3]
```diff
@@ -277,7 +277,12 @@ def string_lower_type(val):
 ARG_NO_BACKFILL = Arg(
     ("--no-backfill",), help="filter all the backfill dagruns given the dag id", action="store_true"
 )
-ARG_STATE = Arg(("--state",), help="Only list the dag runs corresponding to the state")
+ARG_STATE = Arg(
+    ("--state",),
+    help="Only list the dag runs corresponding to the state",
+    metavar="(running, queued, success, failed)",
+    choices=("running", "queued", "success", "failed"),
+)
```
Comment (hussein-awala) [inline airflow/cli/cli_config.py]: Nit: I always have a preference to extract the possible values dynamically when they are based on another class/enum. But I have no objection to defining them this way.
```suggestion
dagrun_states = tuple(state.value for state in DagRunState)
ARG_STATE = Arg(
    ("--state",),
    help="Only list the dag runs corresponding to the state",
    metavar=str(dagrun_states),
    choices=dagrun_states,
)
```

**Review by hussein-awala — APPROVED** [review]
LGTM

File: airflow/cli/cli_config.py (hunk 3 above)
Comment (uranusjr) [reply airflow/cli/cli_config.py]: Instead of `str(dagrun_states)` it’s probably prettier to have `", ".join(dagrun_states)`?

File: airflow/cli/cli_config.py [hunk 4]
```diff
@@ -586,7 +591,6 @@ def string_lower_type(val):
 ARG_CFG_PATH = Arg(("--cfg-path",), help="Path to config file to use instead of airflow.cfg")
 ARG_MAP_INDEX = Arg(("--map-index",), type=int, default=-1, help="Mapped task index")
 
-
```
Comment (uranusjr) [inline airflow/cli/cli_config.py]: Please revert these whitespace changes. If you are interested in changing them, submit a separate pull request.

Comment (rohan472000) [reply airflow/cli/cli_config.py]: by mistake it was there, i have deleted that.

File: airflow/cli/cli_parser.py
Comment (hussein-awala) [inline airflow/cli/cli_parser.py:1]: I think it was deleted by mistake

**Review by hussein-awala — CHANGES_REQUESTED** [review]

File: airflow/cli/cli_parser.py
Comment (rohan472000) [reply airflow/cli/cli_parser.py:1]: Yes I'm adding it again

File: airflow/cli/cli_config.py (hunk 3 above)
Comment (hussein-awala) [reply airflow/cli/cli_config.py]: @uranusjr should it be enclosed in parentheses to align with the other arguments?
```python
metavar=f'({", ".join(dagrun_states)})',
```

**Comment by hussein-awala** [comment]
@rohan472000 there's no need to merge the main branch each time. By doing so, the CI will never be completed to merge this PR

**Review by hussein-awala — APPROVED** [review]
LGTM

**Review by potiuk — APPROVED** [review]

File: airflow/cli/cli_config.py [hunk 5]
```diff
@@ -277,7 +278,13 @@ def string_lower_type(val):
 ARG_NO_BACKFILL = Arg(
     ("--no-backfill",), help="filter all the backfill dagruns given the dag id", action="store_true"
 )
-ARG_STATE = Arg(("--state",), help="Only list the dag runs corresponding to the state")
+dagrun_states = tuple(state.value for state in DagRunState)
+ARG_STATE = Arg(
+    ("--state",),
+    help="Only list the dag runs corresponding to the state",
+    metavar=f'({", ".join(dagrun_states)})',
```
Comment (uranusjr) [inline airflow/cli/cli_config.py]:
```suggestion
    metavar=", ".join(dagrun_states),
```
